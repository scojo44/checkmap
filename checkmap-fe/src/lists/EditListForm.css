.EditListForm {
  border: 2px solid lightgray;
  border-radius: .5rem;
  padding-left: .5rem;
  padding-bottom: .2rem;
}
.EditListForm p {
  margin: .5rem;
}
.EditListForm input[name="description"] {
  display: block;
  width: 95%;
}
