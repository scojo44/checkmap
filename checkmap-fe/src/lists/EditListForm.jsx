import React from 'react'
import {useForm} from 'react-hook-form'
import FormField from '../widgets/FormField';
import './EditListForm.css'

export default function EditListForm({list, update, cancel}) {
  const {register, handleSubmit, formState: {errors}} = useForm({
    defaultValues: {
      name: list.name,
      description: list.description
    }
  });

  return (
    <form className="EditListForm" onSubmit={handleSubmit(async data => await update(data))}>
      <FormField name="name" label="Name" {...{register, errors}} validation={{required: 'Name cannot be blank'}}/>
      <FormField name="description" label="Description (optional)" {...{register, errors}}/>
      <button type="submit">Save</button>
      <button type="button" onClick={cancel}>Cancel</button>
    </form>
  );
}
