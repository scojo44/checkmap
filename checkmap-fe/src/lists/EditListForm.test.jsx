import {render} from '@testing-library/react'
import {MemoryRouter} from 'react-router-dom'
import EditListForm from './EditListForm'

const list = {
  id: 1,
  name: 'Test List 1a',
  description: 'This is a test list.',
  color: '#0000FF',
  regionType: 'State',
  ownerName: 'u1',
  states: [],
  counties: []
}

describe('EditListForm Tests', () => {
  it('Renders without crashing', () => {
    render(
      <MemoryRouter>
        <EditListForm list={list} />
      </MemoryRouter>
    );
  });

  it('Matches snapshot', () => {
    const {asFragment} = render(
      <MemoryRouter>
        <EditListForm list={list} />
      </MemoryRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
