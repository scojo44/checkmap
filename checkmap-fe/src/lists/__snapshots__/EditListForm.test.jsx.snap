// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`EditListForm Tests > Matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="EditListForm"
  >
    <div>
      <p
        class="FormField"
      >
        <label
          for="name"
        >
          Name: 
        </label>
        <input
          name="name"
        />
      </p>
      <p
        class="FormField"
      >
        <label
          for="color"
        >
          Fill Color: 
        </label>
        <input
          name="color"
          type="color"
        />
      </p>
    </div>
    <p
      class="FormField"
    >
      <label
        for="description"
      >
        Description (optional): 
      </label>
      <input
        name="description"
      />
    </p>
    <button
      type="submit"
    >
      Save
    </button>
    <button
      type="button"
    >
      Cancel
    </button>
  </form>
</DocumentFragment>
`;
